File: 找到最大周长的多边形.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        n = len(nums)
        ans = -1
        nums.sort(reverse = True)
        cnt = nums.copy()
        for i in range(1, n):
            cnt[i] += cnt[i - 1]
        for i in range(n):
            if cnt[n - 1] - cnt[i] > nums[i]:
                ans = max(ans, cnt[n - 1] - cnt[i] + nums[i])
        return ans

File: 找到最大周长的多边形.cpp
class Solution {
public:
    long long largestPerimeter(vector<int>& nums) {
        int n = nums.size();
        long long ans = -1;
        sort(nums.begin(), nums.end(), greater());
        vector<long long> cnt(n);
        for (int i = 0; i < n; ++i) {
            cnt[i] = nums[i];
        }
        for (int i = 1; i < n; ++i) {
            cnt[i] += cnt[i - 1];
        }
        for (int i = 0; i < n; ++i) {
            if (cnt[n - 1] - cnt[i] > nums[i]) {
                ans = max(ans, (long long)cnt[n - 1] - (long long)cnt[i] + (long long)nums[i]);
            }
        }
        return ans;
    }
};
